src/modelbench/run.py

```diff
@@ -134,68 +125,27 @@ def find_suts_for_sut_argument(sut_args: List[str]):
-        for h in b.hazards():
-            total += len(h.tests(secrets=secrets))
-    total *= len(suts)
+    benchmark_scores = []
+    for bd, score_dict in run.benchmark_scores.items():
+        for k, score in score_dict.items():
+            benchmark_scores.append(score)
+    return benchmark_scores
 
-    if parallel:
-        with Manager() as manager:
-            shared_count = manager.Value(ctypes.c_double, 0.0)
-            lock = manager.Lock()
-            progress = ProgressTracker(total, json_logs, shared_count, lock)
-            f = functools.partial(score_a_sut, benchmarks, max_instances, secrets, progress, debug)
-            with Pool(len(suts)) as p:
-                results = p.map(f, suts)
-                p.close()
-                p.join()
-            return list(itertools.chain(*results))
-    else:
-        progress = ProgressTracker(total, json_logs)
-        benchmark_scores = []
-        for sut in suts:
-            sut_scores = score_a_sut(benchmarks, max_instances, secrets, progress, debug, sut)
-            benchmark_scores.extend(sut_scores)
-        return benchmark_scores
-
-
-def score_a_sut(benchmarks, max_instances, secrets, progress, debug, sut):
-    sut_scores = []
-    echo(termcolor.colored(f'Examining system "{sut.key}"', "green"))
-    sut_instance = sut.instance(secrets)
-    for benchmark_definition in benchmarks:
-        echo(termcolor.colored(f'  Starting run for benchmark "{benchmark_definition.name()}"', "green"))
-        hazard_scores = []
-        for hazard in benchmark_definition.hazards():
-            echo(termcolor.colored(f'    Examining hazard "{hazard.name()}"', "green"))
-            results = {}
-            for test in hazard.tests(secrets=secrets):
-                items = max_instances
-                results[test.uid] = run_prompt_response_test(
-                    test=test,
-                    sut=sut_instance,
-                    data_dir="./run",
-                    max_test_items=items,
-                    disable_progress_bar=progress.print_updates,  # Proxy for machine-readable logging.
-                )
-                progress.increment()
-
-            score = hazard.score(results)
-            if debug:
-                echo(
-                    termcolor.colored(
-                        f"    For hazard {hazard.name()}, {sut.key} scores {score.score.estimate}", "green"
-                    )
-                )
-            hazard_scores.append(score)
-        benchmark_end_time = datetime.now(timezone.utc)
-        sut_scores.append(benchmark_definition.score(sut, hazard_scores, benchmark_end_time))
-    return sut_scores
+
+def run_benchmarks_for_suts(benchmarks, suts, max_instances, debug=False, json_logs=False, thread_count=32):
+    runner = BenchmarkRunner(pathlib.Path("./run"))
+    runner.secrets = load_secrets_from_config()
+    runner.benchmarks = benchmarks
+    runner.suts = suts
+    runner.max_items = max_instances
```

Comment:
Can `max_instances` default to something sensible based on the machine's capabilities?

Reply:
Speed doesn't vary much based on the user's hardware; it's about the capabilities of service providers like TogetherAI and what the user already has cached. A default here doesn't make much sense, except perhaps `None`, which would mean running everything. 

The higher-up default of 100 for running benchmarks should be about 3 minutes per SUT run, which seems like a good compromise between seeing a useful benchmark and not starting something that takes a week to run.

But the default for calibration should cover all the prompts, because we don't want anybody accidentally recalibrating the standards based on only a fraction of the data.